pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go

```diff
@@ -105,15 +106,16 @@ type ServiceEndpoint struct {
 // the same services. At deployment time all role members will receive
 // identical resource assignments.
 type NodeRole struct {
-	ID            string               `json:"id"`
-	Cardinality   string               `json:"cardinality"`
-	ImageRepoTag  *string              `json:"imageRepoTag,omitempty"`
-	SetupPackage  SetupPackage         `json:"configPackage,omitempty"`
-	PersistDirs   *[]string            `json:"persistDirs,omitempty"`
-	EventList     *[]string            `json:"eventList,omitempty"`
-	MinResources  *corev1.ResourceList `json:"minResources,omitempty"`
-	MinStorage    *MinStorage          `json:"minStorage,omitempty"`
-	ContainerSpec *ContainerSpec       `json:"containerSpec,omitempty"`
+	ID             string               `json:"id"`
+	Cardinality    string               `json:"cardinality"`
+	ImageRepoTag   *string              `json:"imageRepoTag,omitempty"`
+	SetupPackage   SetupPackage         `json:"configPackage,omitempty"`
+	PersistDirs    *[]string            `json:"persistDirs,omitempty"`
+	EventList      *[]string            `json:"eventList,omitempty"`
+	MinResources   *corev1.ResourceList `json:"minResources,omitempty"`
+	MinStorage     *MinStorage          `json:"minStorage,omitempty"`
+	ContainerSpec  *ContainerSpec       `json:"containerSpec,omitempty"`
+	MaxLogSizeDump string               `json:"maxLogSizeDump,omitempty"`
```

Comment:
Assuming we change the type of this to int, then here it will need to be *int so that the validator can tell when it is unset.